server/src/updaters/relationship-updaters.js

```diff
@@ -199,7 +201,65 @@ async function updateRelationships(
 }
 
 function sortIDs(firstId: string, secondId: string) {
-  return [firstId, secondId].map(Number).sort((a, b) => a - b);
+  return [Number(firstId), Number(secondId)].sort((a, b) => a - b);
 }
 
-export { updateRelationships };
+async function knowOfRelationshipsForThread(
+  newMemberIDs: string[],
+  threadID?: string,
+): Promise<RelationshipChangeset> {
+  const combinations = combinations2(newMemberIDs).map(
+    ([firstId, secondId]) => {
+      const [user1, user2] = sortIDs(firstId, secondId);
+      return { user1, user2, status: undirectedStatus.KNOW_OF };
+    },
+  );
+
+  if (threadID) {
+    const query = SQL`
+      SELECT user FROM memberships WHERE thread = ${threadID} ORDER BY user
+    `;
+    const [result] = await dbQuery(query);
+    const memberIDs: number[] = result.map(membership => membership.user);
+
+    return newMemberIDs
+      .reduce((acc, newMemberID) => {
+        return acc.concat(
+          memberIDs.map(memberID => {
+            const [user1, user2] = sortIDs(memberID.toString(), newMemberID);
+            return { user1, user2, status: undirectedStatus.KNOW_OF };
+          }),
+        );
+      }, [])
+      .concat(combinations);
+  } else {
+    return combinations;
+  }
+}
+
+async function updateUndirectedRelationships(
+  changeset: RelationshipChangeset,
+  override: boolean = true,
+) {
+  if (!changeset.length) {
+    return;
+  }
+
+  const rows = changeset.map(row => [row.user1, row.user2, row.status]);
+  const query = SQL`
+    INSERT INTO relationships_undirected (user1, user2, status)
+    VALUES ${rows}
+  `;
+  if (override) {
+    query.append('ON DUPLICATE KEY UPDATE status = VALUES(status)');
```

Comment:
I think it's always a good idea to use `GREATEST` here when setting `know_of` rows, just in case there is a race between a friend request and a new thread being created, for instance. The only time we ever want to replace a `friend` row with a `know_of` row is when the relationship changes (unfriend or block).